easybuild/framework/easyconfig/parser.py

```diff
@@ -89,7 +92,8 @@ def __init__(self, filename=None, format_version=None, rawcontent=None):
 
         self.format_version = format_version
         self._formatter = None
-
+        if filename and os.path.splitext(filename)[-1] == '.yeb':
```

Comment:
don't use `.yeb` all over the place, introduce a constant for it: `YEB_EXTENSION`

and here specifically, I would change this to:

```python
if filename and is_yeb_format(filename):
    ....
```

also, maybe this should move to `_set_formatter`? and we make need a way of detecting whether the `rawcontent` being passed is `.yeb` or `.eb` format (not sure how easy/useful/necessary that is)